### PR TITLE
Jcb/d2iq 72843

### PR DESCRIPTION
**Pull request by JohnB64-JBV**
## Jira Ticket
Before creating this pull request, make sure you have a separate ticket for the docs team to track their work. If you need assistance, ask in the #documentation Slack channel.

https://jira.d2iq.com/browse/D2IQ-72843 

Added Breaking Changes to Release Notes for 2.1.0, 2.1.1, and 2.2, and a short paragraph to the page, "Using a Private Docker Registry" in those versions, as a response to handling the new Docker Hub image pull rate limiting change

## Checklist
- [ ] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [ ] Test all commands and procedures, if applicable.
- [ ] Create your PR against `staging`, not `master`. 
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> ie `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://wiki.d2iq.com/display/ENG/Contributing+to+Docs) for more information.

### CODE REVIEW

File: pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md [hunk 1]
```diff
@@ -39,6 +39,28 @@ DC/OS now allows you to provide a non-CA custom external certificate and key tha
...
+
+  Rate limiting happens on a per-pull basis regardless of whether the pulled image is owned by a paid user. This means D2iQ, as owner of most images used in Mesosphere® DC/OS®, does not have any influence as to whether your current address is rate-limited or not. Mesosphere DC/OS does not have a strict dependency on Docker Hub accounts or plans.
+
+  Without any further configuration, your cluster is likely using the “Free plan - anonymous users” tier. This means that if each of your nodes has its own public IP address, you can do 100 image pulls per 6 hours on every node. While this should not be a problem with just a few, usually-healthy workloads, but you may encounter unforeseeable issues if you:
+
+    - Have constantly failing tasks
+    - Are running a large number of CI jobs
+    - Have metronome tasks with different containers
+    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) command
```
Comment (cneth) [inline pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md:51]: Nit, this is an option, not a 'command'. 

```suggestion
    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) option
```

File: pages/mesosphere/dcos/2.1/release-notes/2.1.1/index.md [hunk 2]
```diff
@@ -18,7 +18,30 @@ DC/OS is a distributed operating system that enables you to manage resources, ap
...
+
+  Rate limiting happens on a per-pull basis regardless of whether the pulled image is owned by a paid user. This means D2iQ, as owner of most images used in Mesosphere® DC/OS®, does not have any influence as to whether your current address is rate-limited or not. Mesosphere DC/OS does not have a strict dependency on Docker Hub accounts or plans.
+
+  Without any further configuration, your cluster is likely using the “Free plan - anonymous users” tier. This means that if each of your nodes has its own public IP address, you can do 100 image pulls per 6 hours on every node. While this should not be a problem with just a few, usually-healthy workloads, but you may encounter unforeseeable issues if you:
+
+    - Have constantly failing tasks
+    - Are running a large number of CI jobs
+    - Have metronome tasks with different containers
+    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) command
```
Comment (cneth) [inline pages/mesosphere/dcos/2.1/release-notes/2.1.1/index.md:31]:
```suggestion
    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) option
```

File: pages/mesosphere/dcos/2.2/release-notes/2.2/index.md [hunk 3]
```diff
@@ -36,6 +36,27 @@ Metronome based jobs can have one or more dependencies specified; a job will onl
...
+
+  Rate limiting happens on a per-pull basis regardless of whether the pulled image is owned by a paid user. This means D2iQ, as owner of most images used in Mesosphere® DC/OS®, does not have any influence as to whether your current address is rate-limited or not. Mesosphere DC/OS does not have a strict dependency on Docker Hub accounts or plans.
+
+  Without any further configuration, your cluster is likely using the “Free plan - anonymous users” tier. This means that if each of your nodes has its own public IP address, you can do 100 image pulls per 6 hours on every node. While this should not be a problem with just a few, usually-healthy workloads, but you may encounter unforeseeable issues if you:
+
+    - Have constantly failing tasks
+    - Are running a large number of CI jobs
+    - Have metronome tasks with different containers
+    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) command
```
Comment (cneth) [inline pages/mesosphere/dcos/2.2/release-notes/2.2/index.md:48]:
```suggestion
    - Use the [docker.forcePullImage](https://mesosphere.github.io/marathon/docs/native-docker.html#forcing-a-docker-pull) option
```

File: pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md [hunk 4]
```diff
@@ -39,6 +39,28 @@ DC/OS now allows you to provide a non-CA custom external certificate and key tha
...
+
+  In the worst case, your cluster might not be able to reschedule a failed task for up to 6 hours, which could lead to unresponsive services or even data corruption, for example, when using clustered databases.
+
+  The DC/OS software offers several ways to specify these credentials:
+
+    1. Cluster-wide credentials using dcos-config.yml: [Using Docker credentials to set cluster-wide registry credentials](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#using-cluster-docker-credentials-to-set-cluster-wide-registry-credentials)
+
+    1. Task-specific credentials using secrets: [Reference private dockcer registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```
Comment (cneth) [inline pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md:59]:
```suggestion
    1. Task-specific credentials using secrets: [Reference private docker registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```

File: pages/mesosphere/dcos/2.1/release-notes/2.1.1/index.md [hunk 5]
```diff
@@ -18,7 +18,30 @@ DC/OS is a distributed operating system that enables you to manage resources, ap
...
+
+  In the worst case, your cluster might not be able to reschedule a failed task for up to 6 hours, which could lead to unresponsive services or even data corruption, for example, when using clustered databases.
+
+  The DC/OS software offers several ways to specify these credentials:
+
+    1. Cluster-wide credentials using dcos-config.yml: [Using Docker credentials to set cluster-wide registry credentials](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#using-cluster-docker-credentials-to-set-cluster-wide-registry-credentials)
+
+    1. Task-specific credentials using secrets: [Reference private dockcer registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```
Comment (cneth) [inline pages/mesosphere/dcos/2.1/release-notes/2.1.1/index.md:39]:
```suggestion
    1. Task-specific credentials using secrets: [Reference private docker registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```

File: pages/mesosphere/dcos/2.2/release-notes/2.2/index.md [hunk 6]
```diff
@@ -36,6 +36,27 @@ Metronome based jobs can have one or more dependencies specified; a job will onl
...
+
+  In the worst case, your cluster might not be able to reschedule a failed task for up to 6 hours, which could lead to unresponsive services or even data corruption, for example, when using clustered databases.
+
+  The DC/OS software offers several ways to specify these credentials:
+
+    1. Cluster-wide credentials using dcos-config.yml: [Using Docker credentials to set cluster-wide registry credentials](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#using-cluster-docker-credentials-to-set-cluster-wide-registry-credentials)
+
+    1. Task-specific credentials using secrets: [Reference private dockcer registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```
Comment (cneth) [inline pages/mesosphere/dcos/2.2/release-notes/2.2/index.md:56]:
```suggestion
    1. Task-specific credentials using secrets: [Reference private docker registry credentials in DC/OS secrets](https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#reference-private-docker-registry-credentials-in-dcos-secrets-enterprise)
```

**Review by cneth — CHANGES_REQUESTED** [review]

**Review by cmastr — APPROVED** [review]

**Review by fatz — APPROVED** [review]
LGTM.

Shall we maybe rename this topic to something like "Docker Registry authentication" and have private docker registries as a subtitle of it?

**Comment by JohnB64-JBV** [comment]
The Vale linter is what triggered the unsuccessful check, and the flagged item is OK.

**Pull request merged**